### PR TITLE
Add nfo parsing for watched, playcount and lastplayed

### LINKED ISSUE #1640: Jellyfin ignores <watched> tags when importing library
**Describe the bug**
When importing an item that has an existing `.nfo` file, the `<watched>` tag is ignored and the item is imported as unwatched.

**To Reproduce**

1. Create a folder called `The 100/Season 01` in a TV library.
1. Place a video file called `The 100 - s01e01 - Pilot.mkv` in that folder.
1. Place the attached `.nfo` file in that folder: [nfofile.zip].(https://github.com/jellyfin/jellyfin/files/3512884/nfofile.zip)
1. Attempt to import the file that has just been added.

**Expected behavior**

The show is imported, and is marked as watched.

**Actual behaviour**
The show is imported, but is marked as unwatched. If Jellyfin is set to write `.nfo` files for that library, the file is changed to contain `<watched>false</watched>`, meaning the user must re-export from Emby before they can try re-importing.

**Logs**

No log errors. (Incidentally, selecting "enable debug logging" does not appear to stick — navigating away from the Logs settings page and back again shows it to be unticked, so if a debug log would help we might have a problem there…)

**Screenshots**

n/a

**System (please complete the following information):**
 - OS: Docker
 - Browser: Firefox
 - Jellyfin Version: 10.3.7
 - Reverse proxy: nginx

**Additional context**

As an Emby library can't be migrated by simply copying the files across, this is essential in order to allow a smooth migration. trakt.tv is not a useful alternative, as involves accepting a massive user agreement and privacy policy from a third party, and then uploading a load of private data for them to monetise. That seems to go against Jellyfin's open source ethos somewhat :)

### PR DESCRIPTION
**Pull request by Ullmie02**
**Changes**
Add support for `watched`, `playcount` and `lastplayed` tag during nfo parsing.
This required adding two interfaced to multiple classes and the tests.
During that I changed some xml comments for the interfaces

**Issues**
Fixes #1640

### CODE REVIEW

**Comment by Ullmie02** [comment]
WIP till I add tests

File: MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs [hunk 1]
```diff
@@ -355,6 +375,50 @@ protected virtual void FetchDataFromXmlNode(XmlReader reader, MetadataResult<T>
...
+
+                        break;
+                    }
+
+                case "playcount":
+                    {
+                        var val = reader.ReadElementContentAsString();
+                        if (!string.IsNullOrWhiteSpace(val) && userData != null)
+                        {
+                            if (int.TryParse(val.Split(' ')[0], NumberStyles.Integer, UsCulture, out var count))
```
Comment (crobibero) [inline MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs:395]: Do some playcounts have spaces in the value?

Comment (Ullmie02) [reply MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs:395]: I hope they don't. Seems like I have copied from runtime which can contain spaces.

**Review by crobibero — APPROVED** [review]

File: MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs [hunk 2]
```diff
@@ -377,40 +377,42 @@ protected virtual void FetchDataFromXmlNode(XmlReader reader, MetadataResult<T>
...
                     }
 
                 case "playcount":
                     {
-                        if (userData != null)
+                        var val = reader.ReadElementContentAsString();
+                        if (!string.IsNullOrWhiteSpace(val) && userData != null)
                         {
-                            userData.PlayCount = reader.ReadElementContentAsInt();
+                            if (int.TryParse(val.Split(' ')[0], NumberStyles.Integer, UsCulture, out var count))
```
Comment (Bond-009) [inline MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs]: Use `IndexOf(' ')` and slice as Span to prevent an allocation

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs (hunk 2 above)
Comment (Ullmie02) [reply MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs]: `playcount` shouldn't contain spaces at all so don't need to split at all

Comment (Bond-009) [reply MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs]: even better

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**